pkg/splunk/enterprise/searchheadcluster_test.go

```diff
@@ -1240,3 +1251,780 @@ func TestGetSearchHeadClusterList(t *testing.T) {
+
+	mclient.AddHandler(wantRequest1, 200, string(response2), nil)
+	mclient.AddHandler(wantRequest2, 200, string(response1), nil)
+	mclient.AddHandler(wantRequest3, 200, string(response3), nil)
+
+	mclient.AddHandler(wantRequest4, 200, string(response2), nil)
+	mclient.AddHandler(wantRequest5, 200, string(response1), nil)
+	mclient.AddHandler(wantRequest6, 200, string(response3), nil)
+
+	mclient.AddHandler(wantRequest7, 200, string(response2), nil)
+	mclient.AddHandler(wantRequest8, 200, string(response1), nil)
+	mclient.AddHandler(wantRequest9, 200, string(response3), nil)
+
+	// mock the verify RF peer funciton
+	VerifyRFPeers = func(ctx context.Context, mgr indexerClusterPodManager, client splcommon.ControllerClient) error {
+		return nil
+	}
+
+	NewSerachHeadClusterPodManager = func(client splcommon.ControllerClient, log logr.Logger, cr *enterpriseApi.SearchHeadCluster, secret *corev1.Secret, newSplunkClient NewSplunkClientFunc) searchHeadClusterPodManager {
+		return searchHeadClusterPodManager{
+			log:     log,
+			cr:      cr,
+			secrets: secret,
+			newSplunkClient: func(managementURI, username, password string) *splclient.SplunkClient {
+				c := splclient.NewSplunkClient(managementURI, username, password)
+				c.Client = mclient
+				return c
+			},
+		}
+	}
+
+	// create directory for app framework
+	newpath := filepath.Join("/tmp", "appframework")
+	err = os.MkdirAll(newpath, os.ModePerm)
+
+	// adding getapplist to fix test case
+	GetAppsList = func(ctx context.Context, s3ClientMgr S3ClientManager) (splclient.S3Response, error) {
+		s3Response := splclient.S3Response{}
+		return s3Response, nil
+	}
+
+	builder := fake.NewClientBuilder()
+	c := builder.Build()
+	utilruntime.Must(enterpriseApi.AddToScheme(clientgoscheme.Scheme))
+	ctx := context.TODO()
+
+	// Create App framework volume
+	volumeSpec := []enterpriseApi.VolumeSpec{
+		{
+			Name:      "testing",
+			Endpoint:  "/someendpoint",
+			Path:      "s3-test",
+			SecretRef: "secretRef",
+			Provider:  "aws",
+			Type:      "s3",
+			Region:    "west",
+		},
+	}
+
+	// AppSourceDefaultSpec: Remote Storage volume name and Scope of App deployment
+	appSourceDefaultSpec := enterpriseApi.AppSourceDefaultSpec{
+		VolName: "testing",
+		Scope:   "local",
+	}
+
+	// appSourceSpec: App source name, location and volume name and scope from appSourceDefaultSpec
+	appSourceSpec := []enterpriseApi.AppSourceSpec{
+		{
+			Name:                 "appSourceName",
+			Location:             "appSourceLocation",
+			AppSourceDefaultSpec: appSourceDefaultSpec,
+		},
+	}
+
+	// appFrameworkSpec: AppSource settings, Poll Interval, volumes, appSources on volumes
+	appFrameworkSpec := enterpriseApi.AppFrameworkSpec{
+		Defaults:             appSourceDefaultSpec,
+		AppsRepoPollInterval: int64(60),
+		VolList:              volumeSpec,
+		AppSources:           appSourceSpec,
+	}
+
+	// create clustermaster custom resource
```

Comment:
@vivekr-splunk can you please comment here. Are we testing the SHC ready state that is also connected with the CM?